sklearn/utils/estimator_checks.py

```diff
@@ -1179,12 +1252,13 @@ def check_classifiers_train(name, classifier_orig):
                     assert_array_equal(np.argmax(decision, axis=1), y_pred)
 
                 # raises error on malformed input for decision_function
-                with assert_raises(ValueError, msg="The classifier {} does"
-                                   " not raise an error when the number of "
-                                   "features in decision_function is "
-                                   "different from the number of features"
-                                   " in fit.".format(name)):
-                    classifier.decision_function(X.T)
+                if not _is_pairwise(classifier):
```

Comment:
Remind me why we don't have the decision_function pairwise case?

Reply:
Or predict_proba

Reply:
Looking at it now. I think my initial reaction was that transposing the pairwise matrix won't raise an error. I'll get it up and running 👍🏽